src/main/java/com/esri/samples/featurelayers/time_based_query/TimeBasedQuerySample.java

```diff
@@ -0,0 +1,132 @@
+
+  private MapView mapView;
+
+  @Override
+  public void start(Stage stage) throws Exception {
+
+    try {
+      // create stack pane and application scene
+      StackPane stackPane = new StackPane();
+      Scene scene = new Scene(stackPane);
+
+      // size the stage, add a title, and set scene to stage
+      stage.setTitle("Time Based Query Sample");
+      stage.setWidth(800);
+      stage.setHeight(700);
+      stage.setScene(scene);
+      stage.show();
+
+      // create a map and set it to a map view
+      mapView = new MapView();
+      ArcGISMap map = new ArcGISMap(Basemap.createOceans());
+      mapView.setMap(map);
+
+      // create a feature table with the URL of the feature service
+      String serviceURL = "https://sampleserver6.arcgisonline.com/arcgis/rest/services/Hurricanes/MapServer/0";
+      ServiceFeatureTable serviceFeatureTable = new ServiceFeatureTable(serviceURL);
+
+      // define the request mode to manual
+      serviceFeatureTable.setFeatureRequestMode(ServiceFeatureTable.FeatureRequestMode.MANUAL_CACHE);
+
+      // load the table and set the query
+      serviceFeatureTable.addDoneLoadingListener(() -> {
+        if (serviceFeatureTable.getLoadStatus() == LoadStatus.LOADED) {
+          // create query parameters
+          QueryParameters queryParameters = new QueryParameters();
+          // set a time extent (beginning of time to September 16th, 2000)
+          Calendar beg = new Calendar.Builder().setDate(1, 1, 1).build();
+          Calendar end = new Calendar.Builder().setDate(2000, 9, 16).build();
+          TimeExtent timeExtent = new TimeExtent(beg, end);
+          queryParameters.setTimeExtent(timeExtent);
+
+          // return all fields
+          List<String> outputFields = Collections.singletonList("*");
+
+          // populate the service with features that fit the time extent, when done zoom to the layer's extent
+          serviceFeatureTable.populateFromServiceAsync(queryParameters, true, outputFields).addDoneListener(() -> {
+            mapView.setViewpointGeometryAsync(serviceFeatureTable.getExtent());
+          });
+
+        } else {
+          new Alert(Alert.AlertType.ERROR, serviceFeatureTable.getLoadError().getMessage()).show();
+        }
+      });
+
+      // create the feature layer using the service feature table
+      FeatureLayer featureLayer = new FeatureLayer(serviceFeatureTable);
+
+      // add the layer to the map
+      map.getOperationalLayers().add(featureLayer);
+
+      // add the map view to stack pane
+      stackPane.getChildren().addAll(mapView);
+
+    } catch (Exception e) {
+      // on any error, display stack trace
+      e.printStackTrace();
+    }
+  }
+
+  /**
+   * Stops and releases all resources used in application.
+   *
+   * @throws Exception if security manager doesn't allow JVM to exit with
+   *           current status
+   */
+  @Override
+  public void stop() throws Exception {
```

Comment:
Isn't this exception not necessary.

Reply:
@john0005 I keep just copying/pasting it ;) The only thing I think it would be useful for is if you want to have a button or something in the app that closes it, and throws an exception if the app fails to stop correctly. I don't think we'll run into that too much with samples so you're right, probably safe to omit it.